internal/modules/module.go

```diff
@@ -285,16 +285,42 @@ func useModuleCache(path string) bool {
 	return true
 }
 
-// ModuleCacheDirectory generates a directory name for the module from the given URI and optional branch.
-func ModuleCacheDirectory(uri, branch string) string {
-	if branch == "" {
-		branch = "v0.0.0"
+// ModuleID returns a unique identifier for a module
+// using the given URI and versioning constraints.
+func ModuleID(uri, version string) string {
+	if version == "" {
+		version = "v0.0.0"
 	}
 
-	return regexp.MustCompile(`[^a-zA-Z0-9<>=.\[\]\-@]+`).ReplaceAllString(uri+"@"+branch, "_")
+	return regexp.MustCompile(`[^a-zA-Z0-9<>=.\[\]\-@]+`).ReplaceAllString(uri+"@"+version, "_")
 }
 
 // StencilCacheDir returns the directory where stencil caches its data.
 func StencilCacheDir() string {
 	return filepath.Join(os.TempDir(), "stencil_cache")
 }
+
+// ModuleCacheDir returns the cache directory for a module based on its type and ID.
+func ModuleCacheDir(cacheType, moduleID string) string {
+	return filepath.Join(StencilCacheDir(), cacheType, moduleID)
+}
+
+// ModuleFSCacheDir returns the cache directory for a module based on its ID.
+func ModuleFSCacheDir(moduleID string) string {
+	return ModuleCacheDir("module_fs", moduleID)
+}
+
+// ModuleVersionCacheDir returns the version cache directory for a module based on its ID.
+func ModuleVersionCacheDir(moduleID string) string {
+	return ModuleCacheDir("module_version", moduleID)
+}
+
+// ModuleVersionLockDir returns the lock directory for a module version based on its ID.
+func ModuleVersionLockDir(moduleID string) string {
+	return ModuleCacheDir("module_version_lock", moduleID)
+}
+
+// ModuleFSLockDir returns the lock directory for a module filesystem based on its ID.
+func ModuleFSLockDir(moduleID string) string {
+	return ModuleCacheDir("module_fs_lock", moduleID)
+}
```

Comment:
issue: I don't know why `golangci-lint` isn't complaining about this in this repo, but typically the linter errors when the naming convention stutters. In this case, it should give a lint error to the effect of `modules.ModuleCacheDir contains module multiple times`.